### PR TITLE
Add div and divrem with rounding argument

### PR DESCRIPTION
**Pull request by paemurru**
We add the standard Julia notation
    `div(5, 2, RoundUp)`
also for the type `ZZRingElem`, so that we can write
    `div(ZZ(5), ZZ(2), RoundUp)`.

Also, we fix a bug. Namely, before there was the line:

```
Base.divrem(x::ZZRingElem, y::Int) = (Base.div(x, y), Base.rem(x, y))
```

But that is inefficient. The whole point of having a `divrem` function is to find the dividend and the remainder together. I have changed the above to

```
Base.divrem(x::ZZRingElem, y::Integer) = Base.divrem(x, ZZ(y))

### CODE REVIEW

**Comment by thofma** [comment]
@lgoettgens since you had a look at those recently, does this make sense? I think so, but it would be good for someone to confirm.

File: src/flint/fmpz.jl
Comment (lgoettgens) [inline src/flint/fmpz.jl:650]:
```suggestion
Base.div(x::Integer, y::ZZRingElem, r::RoundingMode) = Base.div(ZZ(x), y, r)
```

Comment (lgoettgens) [inline src/flint/fmpz.jl:650]:
```suggestion
Base.div(x::Integer, y::ZZRingElem, r::RoundingMode) = Base.div(ZZ(x), y, r)
```
some possible simplification for the redirection versions

File: src/flint/fmpz.jl
Comment (lgoettgens) [inline src/flint/fmpz.jl:658]:
```suggestion
Base.div(x::ZZRingElem, y::Integer, r::RoundingMode) = Base.div(x, ZZ(y), r)
```

File: src/flint/fmpz.jl
Comment (lgoettgens) [inline src/flint/fmpz.jl:667]:
```suggestion
Base.divrem(x::ZZRingElem, y::Integer) = Base.divrem(x, ZZ(y))
```
why not make this oneline like the other ones

File: src/flint/fmpz.jl
Comment (lgoettgens) [inline src/flint/fmpz.jl:676]:
```suggestion
Base.divrem(x::ZZRingElem, y::Integer, r::RoundingMode) = Base.divrem(x, ZZ(y), r)
```

File: src/flint/fmpz.jl
Comment (lgoettgens) [inline src/flint/fmpz.jl:679]:
```suggestion
Base.divrem(x::Integer, y::ZZRingElem) = Base.divrem(ZZ(x), y)
```

**Review by lgoettgens — COMMENTED** [review]
Looks fine mathematically, thanks @paemurru! I just added some suggestions to condense all of the redirection/coercion dispatches with a rounding mode to not consider each and only rounding mode. I know that we don't support all of julias rounding modes, but it will still error once a "real working" function is reached, and it makes it easier to add more rounding modes. Besides that some small formatting suggestions to keep the style more consistent.
Before merging this, I think it would be great if you could add some simple tests that exercise all of these new functions. It doesn't need to be something elaborate or so, but maybe just one pair of inputs that differed between the different rounding modes, and then you input it in different data types and verify that the result is sane.

**Review by lgoettgens — APPROVED** [review]
Thanks, looks good from my POV. Let's wait for CI to finish

**Pull request merged**
